src/Tribe/Data_API.php

```diff
@@ -66,18 +66,23 @@ protected function setup_data() {
 	/**
 	 * Detect what is available in the custom post type by the id passed to it
 	 *
-	 * @param $post_id
+	 * @param null $post id or post object
 	 *
 	 * @return array|bool array includes infomation available and the tribe_tickets_tickets class to use
 	 */
-	public function detect_by_id( $post_id ) {
+
+	public function detect_by_id( $post = null ) {
+
 		// only the rsvp order key is non numeric
-		if ( ! is_numeric( $post_id ) ) {
-			$post_id = esc_attr( $post_id );
-			$cpt     = $this->check_rsvp_order_key_exists( $post_id );
+		if ( ! is_numeric( $post ) ) {
```

Comment:
Won't this also catch objects with an ID property (that would also not be numeric) ... should the clauses be re-ordered?